### PR TITLE
[Feature/ASV-1478] AppView wallet install offer

### PR DESCRIPTION
**Pull request by jdandrade**
**What does this PR do?**

   Shows a promotion offer of the wallet when an user goes into an App with APPC (billing or campaign)

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] AppViewManager.java

**How should this be manually tested?**

Promotion is shown if promotion type is active and wallet is not claimed yet. Check ticket & confluence documentation to view the different states.

  Flow on how to test this or QA Tickets related to this use-case: [confluence doc](https://aptoide.atlassian.net/wiki/spaces/AN/pages/1134559324/APPC+App+Wallet+Offer)

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1478](https://aptoide.atlassian.net/browse/ASV-1478)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/app/view/AppViewFragment.java [hunk 1]
```diff
@@ -264,6 +270,25 @@
...
+  private View walletPromotionButtonsLayout;
+  private Button walletPromotionCancelButton;
+  private Button walletPromotionDownloadButton;
+  private ProgressBar downloadWalletProgressBar;
+  private TextView downloadWalletProgressValue;
+  private ImageView cancelWalletDownload;
+  private ImageView pauseWalletDownload;
+  private ImageView resumeWalletDownload;
+  private PublishSubject<PromotionEvent> promotionAppClick;
+
```
Comment (phillipsgoncalves) [inline app/src/main/java/cm/aptoide/pt/app/view/AppViewFragment.java]: Note: similarAppcAppClick and similarAppcAppsVisibilitySubject are not being used. It is not related to this PR but if you could remove it.. Plus they are not being released on the onDestroy

File: app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java [hunk 2]
```diff
@@ -213,22 +230,39 @@ private void handleOnSimilarAppsVisible() {
...
           crashReport.log(throwable);
         });
   }
 
+  private void handleWalletPromotion() {
+    view.getLifecycleEvent()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(created -> view.isAppViewReadyToDownload())
+        .flatMapSingle(__ -> appViewManager.loadAppViewViewModel())
+        .flatMap(this::updateWalletPromotion)
```
Comment (phillipsgoncalves) [inline app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java]: Don't know if you agree, but I don't agree with this name - updateWalletPromotion. Usually i only call updateXXXX if there is an initial setup and then we are constantly updating that same view, which is not what seems to be happening here.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -633,6 +633,10 @@
   <string name="wallet_promotion_wallet_notinstalled_message">Install this app and get %s APPC Credits. You need the AppCoins Wallet to receive them.</string>
   <string name="wallet_promotion_install_button">Install Wallet</string>
   <string name="wallet_promotion_no_button">No, thanks</string>
+  <string name="wallet_promotion_button_install_disabled">Install to get %s APPC Credits</string>
```
Comment (phillipsgoncalves) [inline app/src/main/res/values/strings.xml]: Shouldn't these strings be added by Carlos? Did these strings went through the correct process?

Comment (jdandrade) [reply app/src/main/res/values/strings.xml]: He knows

**Pull request merged**